File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCStepsDataPoint.java
package com.fjuul.sdk.activitysources.entities.internal.healthconnect;

import java.util.Date;

import androidx.annotation.NonNull;

public class HCStepsDataPoint extends HCDataPoint {
    private final int steps;

    public HCStepsDataPoint(@NonNull Date date, int steps) {
        super(date, null); // Daily data point, no end time
        this.steps = steps;
    }

    public int getSteps() {
        return steps;
    }
} 

File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCStepsDataPoint.kt
package com.fjuul.sdk.activitysources.entities.internal.healthconnect

import java.time.Instant

/**
 * Represents a steps count record from Health Connect.
 */
data class HCStepsDataPoint(
    override val start: Instant,
    override val value: Double,
    override val dataSources: Set<String>
) : HCDailyDataPoint()
